math/algebraic.mbt

```diff
@@ -0,0 +1,96 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/// Compares and returns the maximum of two values.
+///
+/// Returns the second argument if the comparison determines them to be equal.
+///
+/// # Examples
+///
+/// ```
+/// print(@math.maximum(1, 2))  // output: 2
+/// print(@math.maximum(2, 2))  // output: 2
+/// ```
+pub fn maximum[T : Compare](x : T, y : T) -> T {
```

Comment:
The full name `maximum` follows the convention in `list/list.mbt`:

https://github.com/moonbitlang/core/blob/ad0d7c8b12eb3062089a1ccffdb4f3aec462f362/list/list.mbt#L497

However, I personally prefer `max` and `min`, thoughts?